nimare/generate.py

```diff
@@ -0,0 +1,267 @@
+"""Utilities for generating data for testing"""
+import os
+
+import numpy as np
+import nibabel as nib
+
+from .dataset import Dataset
+from .utils import get_resource_path
+from .meta.utils import compute_ma, get_ale_kernel
+from .transforms import vox2mm, mm2vox
+
+DEF_TINY = 1e-50
+
+
+def create_coordinate_dataset(
+    foci_num,
+    fwhm,
+    sample_size,
+    studies,
+    foci_coords=None,
+    foci_noise=0,
+    foci_weights=None,
+    seed=42,
+    sample_size_variance=None,
+):
+    """Generate coordinate based dataset for meta analysis.
+
+    Parameters
+    ----------
+    foci_num : :obj:`int` or :obj:`list`
+        The number of foci to be generated per study
+    fwhm : :obj:`float`
+        Full width at half maximum to define the probability
+        spread of the foci.
+    sample_size : :obj:`int` or :obj:`list`
+        Either mean number of participants in each study
+        or a list specifying the sample size for each
+        study.
+    studies : :obj:`int`
+        Number of studies to generate.
+    foci_coords : :obj:`list` of three item array_like objects (Optional)
```

Comment:
Can you switch from `... (Optional)` to `..., optional` to match `numpydoc`?

Reply:
changed to match numpydoc.